moveit_core/constraint_samplers/src/default_constraint_samplers.cpp

```diff
@@ -475,9 +475,28 @@ bool IKConstraintSampler::samplePose(Eigen::Vector3d& pos, Eigen::Quaterniond& q
     double angle_z =
         2.0 * (random_number_generator_.uniform01() - 0.5) *
         (sampling_pose_.orientation_constraint_->getZAxisTolerance() - std::numeric_limits<double>::epsilon());
-    Eigen::Isometry3d diff(Eigen::AngleAxisd(angle_x, Eigen::Vector3d::UnitX()) *
-                           Eigen::AngleAxisd(angle_y, Eigen::Vector3d::UnitY()) *
-                           Eigen::AngleAxisd(angle_z, Eigen::Vector3d::UnitZ()));
+
+    Eigen::Isometry3d diff;  // (jeroendm) is this second constructor a significant performance penalty?
```

Comment:
That's something we could investigate with @captain-yoshi 's GSoC work. I would delete this comment and instead create an issue listing commits with potential performance impacts.

Reply:
Ok!